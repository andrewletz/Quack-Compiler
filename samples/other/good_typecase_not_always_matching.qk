/**
 * This test case checks that the code functions as expected even when
 * no type case matches.
 */
class Point(x : Int, y:Int) {
    def type_case(other : Obj) : Obj {
        typecase other {
            y : Nothing {
                """Nothing received.\n""".PRINT();
                y.PRINT();
                """\n""".PRINT();
                return;
            }
            z : Int {
                """Int received.\n""".PRINT();
                z.PRINT();
                """\n""".PRINT();
                return;
            }
        }
        """No match found\n""".PRINT();
        return none;
    }
}

pt = Point(7,8);
pt.type_case(pt);

pt.type_case(5);

pt.type_case(none);

pt.type_case(Obj());

pt.type_case(Point(65,33));

