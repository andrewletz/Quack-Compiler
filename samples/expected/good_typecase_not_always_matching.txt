No match found
Int received.
5
Nothing received.
<nothing>
No match found
No match found
